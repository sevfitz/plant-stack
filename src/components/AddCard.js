import React from 'react';
import { connect } from 'react-redux';
import { addCard } from '../store/card.actions';

export function AddCard({ dispatch }) {
    console.log('dispatch is', dispatch);
    return (
        <div>
            <form onSubmit={event => {
                event.preventDefault();
                const form = event.target;
                const { name, genus, species, description, url } = form.elements;
                dispatch(addCard({name: name.value, genus: genus.value, species: species.value, description: description.value, url: url.value}));
                form.reset();
            }}>
                <title>Add A Plant to the Game!</title>
                <label htmlFor="name">Common Name:</label>
                <input id="name" name="name" />

                <label htmlFor="genus">Genus:</label>
                <input id="genus" name="genus" />

                <label htmlFor="species">Species:</label>
                <input id="species" name="species" />

                <label htmlFor="description">Description:</label>
                <input id="description" name="description" />

                <label htmlFor="url">Image url:</label>
                <input id="url" name="url" />

                <button type="submit">Add Plant</button>   
            </form>
        </div>
    );
}

export default connect(
    state => {
        console.log('state is', state);
        return ({ 
            name: state.name, 
            genus: state.genus, 
            species: state.species, 
            description: state.description, 
            url: state.url 
    })}
)(AddCard);
