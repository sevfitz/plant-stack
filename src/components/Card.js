import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

Card.PropTypes = {
    name: PropTypes.string.isRequired,
    genus: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};

export function Choice({ genus, species }) {
    const choiceArray = [genus, species];
    const index = Math.floor(Math.random() * choiceArray.length);
    return (
        <p>
            {choiceArray[index]}
        </p>
    );
}

export function Card({card}) {
    const { name, genus, species, description, url } = card;
    return (
        <div style={{ border: 'solid black 1px', width: 300, height: 400, margin: '50px auto' }}>
            <img style={{ maxHeight: 300 }} src={url} alt={description} />
            <p>Common Name: {name}</p>
            <div><Choice genus={genus} species={species} /></div>
        </div>
    );
}

export default connect(
    state => {
        return ({ 
            name: state.name, 
            genus: state.genus, 
            species: state.species, 
            description: state.description, 
            url: state.url 
    })}
)(Card);

