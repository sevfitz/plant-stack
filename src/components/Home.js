import React from 'react';
import { Link } from 'react-router-dom';

export default function Home() {
    return (
        <div>
            <h2>Welcome to a plant game!</h2>
            <p>Play the game to see a set of plant cards. You'll see a word that is either the genus or species name for the plant. Guess which one it is and click the corresponding button.</p>
            <p>Once you see both the genus and species for a given plant, its card will appear at the bottom. Green text indicates you guessed correctly; red means you were wrong. Play again and again!</p>
            <p>Login to add new cards to the set!</p>
            <Link to="/game">Play the Game!</Link>
        </div>
    )
}