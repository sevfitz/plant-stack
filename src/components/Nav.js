import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { signout } from '../store/auth.actions';
import styled from 'styled-components';

const NavList = styled.ul`
    margin: 0;
    padding: 0;
    `;
    
    const NavItem = styled.li`
    text-align: left;
    display: inline-block;
    margin: 20px;
    list-style-type: none;
`;

const NavLink = props => <Link style={{ color: 'steelblue' }} { ...props}/>;

function Nav({ user, signout }) {
    return (
        <nav>
            <NavList>
                <NavItem><NavLink to="/">Home</NavLink></NavItem>
                <NavItem><NavLink to="/game">Game</NavLink></NavItem>
                <NavItem>
                    { user
                    ? <NavLink to='/' onClick = {signout}>Logout</NavLink>
                    : <NavLink to="/auth/signin">Login</NavLink>
                    /* <NavLink to="/addcard" onClick>Add a Card</NavLink> */
                    }
                </NavItem>
            </NavList>
        </nav>
    );
}

export default connect(
    state => ({ user: state.auth.user }),
    dispatch => ({
        signout() { dispatch(signout()); }
    })
)(Nav);







// import { Link } from 'react-router-dom';
// import { connect } from 'react-redux';
// import { signout } from '../auth/actions';

// const NavLink = props => <Link {...props} />;

// function Nav({ user, signout }) {
//     return (
//         <nav>
//             <ul>
//                 <li><NavLink to="/">Home</NavLink></li>
//                 <li><NavLink to="/game">Game</NavLink></li>
//                 <li>
//                     { user
//                         ? <NavLink to="/" onClick={signout}>Logout</NavLink>
//                         : <NavLink to="/auth/signin">Login</NavLink>
//                     }
//                     </li>
//             </ul>
//         </nav>
//     );
// }

// export default connect(
//     state => ({ user: state.auth.user }),
//     dispatch => ({
//         signout() { dispatch(signout()); }
//     })
// )(Nav);
