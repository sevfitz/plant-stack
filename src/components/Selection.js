import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

const SelectButton = styled.button`
    background-color: #ffed64;
    font-size: 20px;
    height: 40px;
    width: 100px;
    border: 2px solid #ffed64;
    border-radius: 25px;
`;

export function Selection(props) {
    const { card, choice } = props;
    const { _id, name, genus, species } = card;
    return(
        <div>
            <form>
                <p>I think {choice} is the {name}'s...</p>
                <SelectButton>Genus</SelectButton>
                <span> OR </span>
                <SelectButton>Species</SelectButton>
            </form>
        </div>
    );
}

// export default connect(
//     state => {
//         return({
//             name:
//         })
//     }
// )
