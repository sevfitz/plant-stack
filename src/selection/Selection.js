import React from 'react';
// import { connect } from 'react-redux';
import styled from 'styled-components';

// const SelectButton = styled.button`
//     background-color: #ffed64;
//     font-size: 20px;
//     height: 40px;
//     width: 100px;
//     border: 2px solid #ffed64;
//     border-radius: 25px;
// `;

export function Selection(props) {
    const { selectedCard, selection, cards, onSelectGenus, onSelectSpecies } = props;
    const { _id, choice } = selection;
    
    

    return(
        <form className="card-footer" onSubmit={event => {
            event.preventDefault();
        }}>
            {/* <span className="card-footer-item"> */}
                <button className="button is-medium is-outlined is-primary is-pulled-left" type="submit" onClick={() => onSelectGenus(_id, choice, cards)}>Genus</button>
            {/* </span> */}
            {/* <span className="card-footer-item"></span> */}
            {/* <span className="card-footer-item"> */}
                <button className="button is-medium is-outlined is-primary is-pulled-right" type="submit" onClick={() => onSelectSpecies(_id, choice, cards)}>Species</button>
            {/* </span> */}
        </form>
    );
}

// export default connect(
//     state => {
//         return ({ 
//             name: state.name, 
//             genus: state.genus, 
//             species: state.species, 
//             description: state.description, 
//             url: state.url,
//             seenCards: state.seenCards,
//             cards: state.cards
//     })}
// )(Selection);

// const mapStateToProps = (state) => {
//     const selectedCard = state.cards.find((card) => card._id === state.selection._id);
//     return {
//         card: selectedCard,
//         name: state.name, 
//         genus: state.genus, 
//         species: state.species, 
//         description: state.description, 
//         url: state.url,
//         seenCards: state.seenCards,
//         cards: state.cards
//     };
// };

// function mapDispatchToProps(dispatch) {
//     return bindActionCreators({ chooseGenus, chooseSpecies }, dispatch);
// }

// const mapDispatchToProps = {
//     chooseGenus,
//     chooseSpecies
// }

// export default connect(mapStateToProps, mapDispatchToProps)(Selection);
