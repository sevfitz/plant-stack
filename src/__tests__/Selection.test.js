import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Selection } from '../components/Selection';

describe('Selection Component Snapshot', () => {

    it('displays correct properties', () => {
        const wrapper = shallow(<Selection />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});


// describe('Choice Component Test', () => {

//     it('displays correct properties', () => {
//         const wrapper = shallow(<Choice genus='' species='' />);
//         expect(toJSON(wrapper)).toMatchSnapshot();
//     });
// });

// describe('Card Component Test', () => {

//     it('displays correct properties', () => {
//         const wrapper = shallow(<Card name={name} genus={genus} species={species} description={description} url={url} />);
//         expect(toJSON(wrapper)).toMatchSnapshot();
//     });

// });