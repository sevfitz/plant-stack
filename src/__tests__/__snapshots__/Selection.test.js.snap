// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection Component Snapshot displays correct properties 1`] = `
<div>
  <form>
    <p>
      I think this is this plant's...
    </p>
    <styled.button>
      Genus
    </styled.button>
    <styled.button>
      Species
    </styled.button>
  </form>
</div>
`;
